assets/app/js/componentDialog.js

```diff
@@ -151,9 +159,29 @@ class ComponentDialog {
         $('select').trigger('change.select2');
 
         this.title.text(data.id);
+
+        // Disable the ability to close the dialog by clicking outside the dialog or pressing the escape key.
+        this.dialog.modal({ backdrop: "static", keyboard: false });
+
+        // Retrieve any previously-saved values -- use below in _closeComponent() to compare with any new edits before closing the dialog.
+        this.initialSerialization = this.dialog.find("form").serialize();
+
         this.dialog.modal('show');
     }
 
+    // Check whether the user has made any new edits.
+    _closeComponent() {
+        // Retrieve the current form values, i.e., including edits not yet saved.
+        this.currentSerialization = this.dialog.find("form").serialize();
+
+        if (this.initialSerialization !== this.currentSerialization) {
+            if(!confirm('Any edits you\'ve made will be lost if you close without saving.  \n\nClick "OK" if you don\'t want to save your edits.')) {
```

Comment:
I'm wondering if we should be more explicit with the button text. For example:

![component confirmation save](https://user-images.githubusercontent.com/4239800/33624459-54621be4-d9a9-11e7-947f-667079697ef8.png)

Reply:
@jdaguil  I like your suggestion.  It appears that the default confirmation dialog displays only `OK` and `Cancel` and cannot be customized.  Shall I explore creating your more-informative buttons with Bootstrap or perhaps as a jQuery Dialog?  

And if so, we have one or more other similar dialogs -- would we want those to have a similar look-and-feel as well?

Reply:
@johnmhoran I think this can be explored in a new issue, and we can go ahead with the feature work you've done in this PR.

Reply:
@jdaguil  Excellent.  Coming right up . . . .